src/EnergyPlus/Boilers.cc

```diff
@@ -100,12 +100,12 @@ PlantComponent *BoilerSpecs::factory(EnergyPlusData &state, std::string const &o
         GetBoilerInput(state);
         state.dataBoilers->getBoilerInputFlag = false;
     }
-    // Now look for this particular pipe in the list
-    for (auto &boiler : state.dataBoilers->Boiler) {
-        if (boiler.Name == objectName) {
-            return &boiler;
-        }
-    }
+    // Now look for this particular boiler in the list
+    auto myBoiler = std::find_if(state.dataBoilers->Boiler.begin(), state.dataBoilers->Boiler.end(), [&objectName](const BoilerSpecs &boiler) {
+        return boiler.Name == objectName;
+    });
+    if (myBoiler->Name == objectName) return myBoiler;
+
```

Comment:
Need comments before this is used elsewhere. CppCheck suggestion:

    [src/EnergyPlus/Boilers.cc:105]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.

Reply:
Sorry, happy finger. Same comment here.